rivertest/worker.go

```diff
@@ -0,0 +1,223 @@
+package rivertest
+
+import (
+	"context"
+	"encoding/json"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/riverqueue/river"
+	"github.com/riverqueue/river/internal/dbunique"
+	"github.com/riverqueue/river/internal/execution"
+	"github.com/riverqueue/river/riverdriver"
+	"github.com/riverqueue/river/rivershared/baseservice"
+	"github.com/riverqueue/river/rivershared/testfactory"
+	"github.com/riverqueue/river/rivershared/util/ptrutil"
+	"github.com/riverqueue/river/rivershared/util/sliceutil"
+	"github.com/riverqueue/river/rivershared/util/valutil"
+	"github.com/riverqueue/river/rivertype"
+)
+
+// Worker makes it easier to test river workers. Once built, the worker can be
+// used to work any number of synthetic jobs without touching the database:
+//
+//	worker := rivertest.NewWorker(t, driver, config, worker)
+//	if err := worker.Work(ctx, t, args, nil); err != nil {
+//		t.Fatalf("failed to work job: %s", err)
+//	}
+//
+//	if err := worker.Work(ctx, t, args, &river.InsertOpts{Queue: "custom_queue"}); err != nil {
+//		t.Fatalf("failed to work job: %s", err)
+//	}
+//
+// In all cases the underlying [river.Worker] will be called with the synthetic
+// job. The execution environment has a realistic River context with access to
+// all River features, including [river.ClientFromContext] and worker
+// middleware.
+//
+// When relying on features that require a database record (such as JobCompleteTx),
+// the job must be inserted into the database first and then executed with
+// WorkJob.
```

Comment:
We can probably punt on this, but it'd probably make sense to have a test or something that'd make sure users have a sane way to do this.

`JobCompleteTx` will probably refuse to complete the job unless it's `running`, and there might not be that easy of a way to manipulate a job from `available` to `running` right now. They could `Client.Insert`, but then it'd get stuck there.

A couple possibilities:

* We could provide a test helper for inserting jobs in any state similar to our internal factories.
* We could have the test helpers here check to see if a job exists in the DB when running. If so, transition the job to `running`.

Reply:
Yeah, I just ran into that while writing the test for this! 😆 I think it would make sense as a fast follow so we can refine the API a little more.